src/gennylib/CMakeLists.txt

```diff
@@ -16,143 +16,16 @@
 # src/CMakeList.txt, src/gennylib/CMakeLists.txt and src/gennylib/src/version.cpp
 project(gennylib VERSION 0.0.1 LANGUAGES CXX)
 
-#Pacify the CLion
-file(GLOB_RECURSE HEADERS_FOR_CLION
-     RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
-     CONFIGURE_DEPENDS
-     include/*.hpp include/*.hh src/*.hh)
-add_library(gennylib_headers SHARED
-    EXCLUDE_FROM_ALL
-    src/version.cpp
-    ${HEADERS_FOR_CLION}
+genny_subdir(
+    NAME    gennylib
+    TYPE    STATIC
+    DEPENDS
+        yaml-cpp
+        loki
+        metrics
+        value_generators
+        Boost::boost
+        Boost::log
+        MongoCxx::mongocxx
+    TEST_DEPENDS    testlib
 )
-
-add_library(gennylib_core STATIC
-    src/version.cpp
-)
-target_include_directories(gennylib_core
-    PUBLIC
-        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
-        $<INSTALL_INTERFACE:include>
-)
-target_link_libraries(gennylib_core
-    yaml-cpp
-    loki
-    metrics
-    value_generators
-    Boost::boost Boost::log
-    MongoCxx::mongocxx
-)
-
-# Concepts to do with Actors
-add_library(gennylib_cast STATIC
```

Comment:
I *think* there were a few sub-targets within `gennylib` in order to maybe improve compile-time and maybe to keep some things separate that probably didn't need to be together, is that right @bcaimano?

The new `genny_subdir` thing isn't quite so flexible and would force us to create separate cmake projects for each of these sub-targets - do you think that's worth it? My instinct is no but open to ideas here since I'm not sure of all the reasons why this split was added (I was out while it went in).